resources/sdk/clisdkclient/templates/restclient.mustache

```diff
@@ -0,0 +1,197 @@
+
+	"github.com/tidwall/pretty"
+)
+
+var (
+	Client HTTPClient
+)
+
+// HTTPClient interface
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type HttpStatusError struct {
+	Verb       string
+	Path       string
+	Body       string
+	StatusCode int
+	Headers    map[string][]string
+}
+
+func (e HttpStatusError) Error() string {
+	return fmt.Sprintf("HttpStatusCode error encountered while processing request to Genesys Cloud API. Verb %s, Path %s, Status Code %d, Body %v", e.Verb, e.Path, e.StatusCode, e.Body)
+}
+
+type RESTClient struct {
+	environment string
+	token       string
+}
+
+type oAuthToken struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   string `json:"expires_in"`
+	Error       string `json:"error"`
+}
+
+//Get executes an HTTP Get
+func (r *RESTClient) Get(uri string) (string, error) {
+	return r.callAPI(http.MethodGet, uri, "")
+}
+
+//Put executes an HTTP Put
+func (r *RESTClient) Put(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPut, uri, data)
+}
+
+//Post executes an HTTP Post
+func (r *RESTClient) Post(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPost, uri, data)
+}
+
+//Patch executes an HTTP Patch
+func (r *RESTClient) Patch(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPatch, uri, data)
+}
+
+//Delete executes an HTTP Delete
+func (r *RESTClient) Delete(uri string) (string, error) {
+	return r.callAPI(http.MethodDelete, uri, "")
+}
+
+func (r *RESTClient) callAPI(method string, uri string, data string) (string, error) {
+	var apiURI *url.URL
+	if !strings.HasPrefix(uri, "/") {
+		uri = fmt.Sprintf("/%v", uri)
+	}
+	if !strings.Contains(r.environment, "localhost") {
+		apiURI, _ = url.Parse(fmt.Sprintf("https://api.%s%s", r.environment, uri))
+	} else {
+		apiURI, _ = url.Parse(fmt.Sprintf("http://%s%s", r.environment, uri))
+	}
+
+	request := &http.Request{
+		URL:    apiURI,
+		Close:  true,
+		Method: strings.ToUpper((method)),
+		Header: make(map[string][]string),
+	}
+
+	//Setting up the auth header
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", r.token))
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Cache-Control", "no-cache")
+
+    //User-Agent and SDK version headers
+    request.Header.Set("User-Agent", "{{#httpUserAgent}}{{.}}{{/httpUserAgent}}{{^httpUserAgent}}Swagger-Codegen{{/httpUserAgent}}/go-cli")
+    request.Header.Set("purecloud-sdk", "{{packageVersion}}")
+
+	if data != "" {
+		request.Body = ioutil.NopCloser(bytes.NewBuffer([]byte(data)))
+	}
+
+	//Executing the request
+	resp, err := Client.Do(request)
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	responseData := string(response)
+
+	if (resp.StatusCode != http.StatusOK) && (resp.StatusCode != http.StatusAccepted) {
```

Comment:
Should this check for any 2xx status code? Maybe do a check like `if resp.StatusCode < 200 || resp.StatusCode >=300`. If not, it should at least also check for StatusCreated (201) and StatusNoContent (204) as those are both used in the API.

Reply:
I think thats a good idea.  When I originally wrote the good I wrote because I only ever dealt with 200 and 202s.  I think we should with your first comment Tim.  If our APIs support other "Success" status code then we should implement this.